crates/sui-types/src/transaction.rs

```diff
@@ -3354,10 +3366,35 @@ pub enum InputObjectKind {
     SharedMoveObject {
         id: ObjectID,
         initial_shared_version: SequenceNumber,
-        mutable: bool,
+        mutability: SharedObjectMutability,
     },
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize, PartialOrd, Ord, Hash)]
+pub enum SharedObjectMutability {
+    // The "classic" mutable/immutable modes.
+    Immutable,
+    Mutable,
+}
+
+impl std::fmt::Display for SharedObjectMutability {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            SharedObjectMutability::Mutable => write!(f, "Mutable"),
+            SharedObjectMutability::Immutable => write!(f, "Immutable"),
+        }
+    }
+}
+
+impl SharedObjectMutability {
+    pub fn is_mutable(&self) -> bool {
+        match self {
+            SharedObjectMutability::Mutable => true,
+            SharedObjectMutability::Immutable => false,
+        }
+    }
```

Comment:
Is this something we want to support? What happens after the non-exclusive variant is added?